File: app/src/main/java/com/popalay/cardme/injection/AppComponent.java
package com.popalay.cardme.injection;

import com.popalay.cardme.presentation.screens.addcard.AddCardPresenter;
import com.popalay.cardme.presentation.screens.adddebt.AddDebtPresenter;
import com.popalay.cardme.presentation.screens.cards.CardsPresenter;
import com.popalay.cardme.presentation.screens.cards.CardsViewModel;
import com.popalay.cardme.presentation.screens.debts.DebtsPresenter;
import com.popalay.cardme.presentation.screens.holderdetails.HolderDetailsActivity;
import com.popalay.cardme.presentation.screens.holderdetails.HolderDetailsViewModel;
import com.popalay.cardme.presentation.screens.holders.HoldersPresenter;
import com.popalay.cardme.presentation.screens.settings.SettingPresenter;
import com.popalay.cardme.presentation.screens.splash.SplashPresenter;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {

    void inject(AddCardPresenter presenter);

    void inject(CardsPresenter presenter);

    void inject(HoldersPresenter presenter);

    void inject(AddDebtPresenter presenter);

    void inject(DebtsPresenter presenter);

    void inject(SettingPresenter presenter);

    void inject(CardsViewModel viewModel);

    void inject(HolderDetailsViewModel viewModel);

    void inject(HolderDetailsActivity activity);

    void inject(SplashPresenter presenter);
}

File: app/src/main/java/com/popalay/cardme/injection/AppComponent.kt
package com.popalay.cardme.injection

import com.popalay.cardme.App
import com.popalay.cardme.presentation.base.navigation.CustomRouter
import com.popalay.cardme.presentation.screens.adddebt.AddDebtPresenter
import com.popalay.cardme.presentation.screens.debts.DebtsPresenter
import com.popalay.cardme.presentation.screens.home.HomePresenter
import com.popalay.cardme.presentation.screens.splash.SplashPresenter
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton


@Singleton
@Component(modules = arrayOf(
        AndroidSupportInjectionModule::class,
        UiModule::class,
        AppModule::class)
)
interface AppComponent : AndroidInjector<App> {

    @Component.Builder
    interface Builder {
        @BindsInstance fun application(application: App): Builder
        fun build(): AppComponent
    }

    fun getRouter(): CustomRouter

    fun inject(presenter: AddDebtPresenter)

    fun inject(presenter: DebtsPresenter)

    fun inject(presenter: SplashPresenter)

    fun inject(presenter: HomePresenter)
}